src/Filter.tsx

```diff
@@ -0,0 +1,337 @@
+    Background: string | undefined;
+    Slug: string | undefined;
+    Chest: string | undefined;
+    Mouth: string | undefined;
+    Head: string | undefined;
+    Eyes: string | undefined;
+    Tail: string | undefined;
+    Back: string | undefined;
+    Hands: string | undefined;
+    [key: string]: string | undefined;
+}
+
+export interface FilterProps {
+    traitNameMap?: Map<string, Trait>;
+
+    attributes?: Attribute[];
+
+    onChange: (data: Traits) => void;
+}
+
+export interface AttributeProps {
+    trait?: Trait;
+
+    attributes?: Attribute[];
+
+    attribute: string;
+
+    traitNameMap?: Map<string, Trait>;
+
+    showTraitRarity: boolean;
+
+    onChange: (newTrait: string) => void;
+}
+
+function AttributeWrapper(props: AttributeProps) {
+    const {
+        attributes,
+        attribute,
+        trait,
+        traitNameMap,
+        showTraitRarity,
+        onChange,
+    } = props;
+
+    const attributeDivClasses = showTraitRarity ? "w-40" : "w-56";
+    const attributeDivBigClasses = showTraitRarity ? "w-56" : "w-64";
+
+    const dropdownWidth = showTraitRarity ? "w-40" : "w-52";
+
+    const availableValues = React.useMemo(() => {
+        if (!attributes) {
+            return [];
+        }
+
+        for (const attr of attributes) {
+            if (attr.name === attribute) {
+                const options = attr.values.reduce((accumulator, a) => {
+                    if (a.name !== "None") {
+                        accumulator.push({
+                            value: a.name,
+                            label: a.name,
+                        });
+                    }
+
+                    return accumulator;
+                }, [] as { value: string; label: string }[]);
+
+                options.sort((a, b) => a.label.localeCompare(b.label));
+
+                options.unshift({
+                    value: "",
+                    label: "None",
+                });
+
+                return options;
+            }
+        }
+
+        return [];
+    }, [attributes, attribute]);
+
+    return (
+        <div
+            className={`flex flex-col items-start justify-center ${attributeDivClasses} 2xl:${attributeDivBigClasses}`}
+            key={attribute}
+        >
+            <Dropdown
+                label={attribute}
+                value={trait?.name || ""}
+                onChange={onChange}
+                options={availableValues}
+                headerStyle={`text-2xl text-slugGreen uppercase`}
+                valueStyle={`text-4xl uppercase truncate`}
+                align={Align.Start}
+                dropdownWidth={dropdownWidth}
+            />
+
+            {showTraitRarity && trait && (
+                <span className="text-2xl uppercase">
+                    {traitNameMap!.get(`${attribute}-${trait.name}`)!.rarity}
+                </span>
+            )}
+        </div>
+    );
+}
+
+
+const Filter: React.FC<FilterProps> = (props: FilterProps) => {
```

Comment:
Lets export this as a regular function

```suggestion
export function Filter(props: FilterProps) {
```